Source/ThirdParty/ANGLE/src/libANGLE/Framebuffer.cpp

```diff
@@ -1812,8 +1810,6 @@ angle::Result Framebuffer::invalidate(const Context *context,
     // Back-ends might make the contents of the FBO undefined. In WebGL 2.0, invalidate operations
     // can be no-ops, so we should probably do that to ensure consistency.
     // TODO(jmadill): WebGL behaviour, and robust resource init behaviour without WebGL.
-    if (context->isRobustResourceInitEnabled())
-        markDrawAttachmentsNeedInit(count, attachments);
```

Comment:
This will break some WebXR content. An explicit clear will need to be added back into WebXROpaqueFramebuffer too.

Reply:
Maybe we can mark the attachment WebXROpaqueFramebuffer uses as needing to take this path, as there is the expectation the clear occurs there, without adding the manual clear back?

I will give that a try

Reply:
Is it valid to only call `markDrawAttachmentsNeedInit` from the `discardFramebuffer` path (which WebXR uses)?

I updated the patch to do that to avoid causing a performance regression in WebXR content. Maybe there is a better way to do it.

Reply:
> Is it valid to only call markDrawAttachmentsNeedInit from the discardFramebuffer path (which WebXR uses)?

That was the intention. Internally, can we verify that the discard path is only used for WebXR and not for the WebGL2 Invalidate framebuffer?

It would be safer to put an explicit glClear into the `WebXROpaqueFramebuffer::startFrame`, or modify ANGLE to replace `MTLLoadActionDontCare` with `MTLLoadActionClear` when robust initialisation is enabled on the context. (This is my preferred solution)

Reply:
Sounds good, I will do that instead